### ISSUE TITLE
Consider magically transposing points coordinates if tuple input.

### ISSUE THREAD

**Issue by jni** [napari/napari#681]
## 🚀 Feature

Some functions, such as NumPy's `np.nonzero` (and NumPy fancy indexing in general) and `scipy.ndimage.map_coordinates` use (D, N) as the shape of the coordinates array, instead of `(N, D)`. This means that if you have 34 3D points in this format, `viewer.add_points` will try to add 3 34-dimensional points. At this point it actually crashes.

Since users might be passing the output of `np.nonzero` to `add_points` with some frequency, it might be advisable to check if the type is a tuple, or even more specifically, a tuple of integer 1D arrays, and transpose automatically in that case.

## Alternatives

If we decide *not* to be so magical, we might want to have some maximum dimensionality, beyond which we refuse to add the points unless given a `force=True` parameter, for example, and ask the user whether they are sure they want to add these points.

## Additional context

First observed by @stefanv in #467

**Comment by stefanv** [comment]
Question: should the dimensionality of the points added not always match the data displayed? E.g. it seems as though adding 32-dimensional data should fail because we are visualizing 2D or 3D slices.

**Comment by jni** [comment]
@stefanv no, because napari can do broadcasting, even of different layer types. See [this video](https://ilovesymposia.com/napari/napari-hmaxima-bug.mp4) showing 3D points on a 2D image. (From the "parameter sweeps" section of the [blog post](https://ilovesymposia.com/2019/10/24/introducing-napari-a-fast-n-dimensional-image-viewer-in-python/).) (Sorry, my Nikola config doesn't yet autogenerate section links!)
